packages/flutter_tools/test/general.shard/project_test.dart

```diff
@@ -1568,11 +1589,13 @@ void addIosProjectFile(Directory directory, {required String Function() projectF
     ..writeAsStringSync(projectFileContent());
 }
 
-void addAndroidGradleFile(Directory directory, { required String Function() gradleFileContent }) {
+void addAndroidGradleFile(Directory directory, {
```

Comment:
Add a dart doc that says that kotlinDsl true does not actually use the kotlin dsl but instead names the file with the kotlin dsl. I do not think this cl needs to add all the kotlin dsl functionality but it should warn the next caller of this method. When we add branching logic, for example different regex, then we can update this method and the dartdoc.

Reply:
done in 1c22b6f10feb61e75e0459c7876f4f9207112b29

Reply:
Not quite, Can you make the dart doc *warn* the caller that addAndroidGradleFile does not actually build a kotlin dsl gradle file but instead just changes the file extension. 

I think asking you to modify the test code to actually write the kotlin dsl is more than is required but future test callers should at least be warned that if they need to test the behavior difference in the content that they need to update this helper.

Reply:
ah okay, I get you now

fixed in 8e696799f871ddf65cf1d37bf484512ace857cae